File: web/src/types/audit.ts
export interface AuditDiff {
  field: string;
  old_value: unknown;
  new_value: unknown;
}

export interface AuditLog {
  id: string;
  tenant_id: string;
  user_id: string;
  user_name: string;
  user_email: string;
  action: string;
  resource_type: string;
  resource_id: string;
  resource_name?: string;
  ip_address?: string;
  user_agent?: string;
  diff?: AuditDiff[];
  before_state?: Record<string, unknown>;
  after_state?: Record<string, unknown>;
  timestamp: string;
}

export interface AuditFilter {
  action?: string;
  resource_type?: string;
  user_id?: string;
  start_date?: string;
  end_date?: string;
}

File: backend/internal/domain/audit.go
package domain

import "time"

type AuditLog struct {
	ID           string                 `json:"id"`
	TenantID     string                 `json:"tenant_id"`
	ActorUserID  string                 `json:"actor_user_id,omitempty"`
	ActorEmail   string                 `json:"actor_email,omitempty"`
	Action       string                 `json:"action"`
	ResourceType string                 `json:"resource_type"`
	ResourceID   string                 `json:"resource_id,omitempty"`
	Before       map[string]interface{} `json:"before,omitempty"`
	After        map[string]interface{} `json:"after,omitempty"`
	IPAddress    string                 `json:"ip_address,omitempty"`
	CreatedAt    time.Time              `json:"created_at"`
}
